### ISSUE TITLE
Can't turn a permalink into an object for 11ty Serverless

### ISSUE THREAD

**Issue by MichaelSolati** [11ty/eleventy#1898]
**Describe the bug**
At the top level of my 11ty project I've created a `11tydata.js` file that sets the permalink of all pages of my project: like this...

```js
module.exports = {
  eleventyComputed: {
    permalink: (data) => data.permalink.replace(
      /^\/{{lang}}/,
      some_condition ? '/i18n/{{lang}}' : '',
    ),
  },
};
```

This works fine, but I'm interested in running the whole project in a serverless environment so I thought instead of updating close to 1000 md/njk files to have:

```md
---
permalink:
  serverless:  '/design-system/colors/index.html'
---
```

I could instead have the catch all permalink changer look like this...

```js
module.exports = {
  eleventyComputed: {
    permalink: (data) => ({
      serverless: data.permalink.replace(
        /^\/{{lang}}/,
        some_condition ? '/i18n/{{lang}}' : '',
      ),
    }),
  },
};
```

However I end up getting the following error:

```
Problem writing Eleventy templates: (more in DEBUG output)
> Having trouble writing template: false

`TemplateWriterWriteError` was thrown
> The "path" argument must be of type string. Received type boolean (false)
```

If I try this:

```js
module.exports = {
  eleventyComputed: {
    permalink: {
      serverless: (data) =>
        data.permalink.replace(/^\/{{lang}}/, some_condition ? '/i18n/{{lang}}' : ''),
    },
  },
};
```

Then the `data.permalink` returns an object rather than a string and the code also fails...

```
Problem writing Eleventy templates: (more in DEBUG output)
> data.permalink.replace is not a function
```

**To Reproduce**

See above...

**Expected behavior**
I should be able to `eleventyComputed` a `permalink` and return an object, not just a string.

**Environment:**

- Linux
- Eleventy Version 1.0.0-canary.39

**Comment by zachleat** [comment]
A few things I noticed here just to get started:

1. It looks like template and directory data files aren’t being bundled. This is a _bug_. 😱 You can workaround this by setting your permalink switcher in an `eleventyComputed.js` global data file.
2. Can you always assume that `data.permalink` in your computed data callback is actually a string and always exists? Can you re-run your tests without `replace` and see if it works? You may want to check for existence and type before using in this way

**Comment by zachleat** [comment]
I was able to confirm that `_data/eleventyComputed.js` works for:

```js
module.exports = {
  permalink: function(data) {
    return {
      serverless: `${data.page.filePathStem}/`
    };
  }
}
```

and

```js
module.exports = {
  permalink: {
    serverless: data => `${data.page.filePathStem}/`
  }
}
```

**Comment by MichaelSolati** [comment]
This was a major step forward, but it seems like paginated pages continue to write false... I'll create a simplified repo where this issue persists.

**Comment by zachleat** [comment]
Related fix for issue #1911 will ship with 1.0.0 canary 40

**Comment by zachleat** [comment]
Just want to cross-link the different (subtle, but different) bugs that were fixed today in #2222 and #2288

**Comment by zachleat** [comment]
I added a bunch of tests to confirm that this is now working. If you’d like to add/modify them, please check the commits and open a PR!

https://github.com/11ty/eleventy/commit/dd99adc1ac13d09770d12fc5533821cf29f6a8bf
https://github.com/11ty/eleventy/commit/72731303dfef6d212b5c640a489fb39c87107e15

> This is an automated message to let you know that a helpful response was posted to your issue and for the health of the repository issue tracker the issue will be closed. This is to help alleviate issues hanging open waiting for a response from the original poster.

> If the response works to solve your problem—great! But if you’re still having problems, do not let the issue’s closing deter you if you have additional questions! Post another comment and we will reopen the issue. Thanks!
